## Problem Statement

An e-commerce company is evaluating two different website designs to see which one results in higher customer engagement. Design A is the current design, while Design B incorporates new features aimed at improving user experience. The company hypothesizes that Design B will lead to a higher average time spent on the website by users.

**Datasets:**
- current_design.csv: Contains data for user interactions with the current website design (Design A), with columns user_id and time_spent_minutes.
- new_design.csv: Contains data for user interactions with the new website design (Design B), with columns user_id and time_spent_minutes.

**Objective:**
- To determine whether Design B results in a higher average time spent on the website compared to Design A.

**Steps to perform:**
- Set the null and alternate hypothesis for this analysis.
- Load the datasets current_design.csv and new_design.csv.
- Calculate the mean and standard deviation of the time spent for both designs.
- Determine the sizes of both groups.
- Calculate the z-score to compare the means of both groups.
- Set the significance level (alpha) at 5% for a right-tailed test.
- Calculate the critical z-value for the right-tailed test at the 5% significance level.
- Compare the calculated z-score with the critical z-value to decide whether to reject the null hypothesis.
- Write down your observations in the end.

**Import Necessary Libraries**

In [3]:
import pandas as pd
import numpy as np
from scipy import stats as st

**Define hypothesis**



H0 - design B doesnt lead to higher time spend
H1 - design B leads to higher time spend

**1: Load the datasets**

In [6]:
df_a = pd.read_csv("..\\datasets\\current_design.csv")
df_b = pd.read_csv("..\\datasets\\new_design.csv")

df_b

user_id  time_spent_minutes
0     T001                7.49
1     T002                7.37
2     T003                7.32
3     T004                6.85
4     T005                7.10
..     ...                 ...
95    T096                8.27
96    T097                7.79
97    T098                8.99
98    T099                7.12
99    T100                8.79

[100 rows x 2 columns]

**2: Calculate the mean and standard deviation of the time spent for both designs.**

In [7]:
#control statistics
mean_a = df_a["time_spent_minutes"].mean()
std_a = df_a["time_spent_minutes"].std()
mean_a, std_a

(6.015199999999998, 0.6182550877553322)

In [8]:
#test statistics
mean_b = df_b["time_spent_minutes"].mean()
std_b = df_b["time_spent_minutes"].std()
mean_b, std_b

(8.062599999999998, 0.9025257711981236)

**3: Test using rejection region (i.e. critical z value)**

In [11]:
a = std_a**2/df_a.shape[0]
b = std_b**2/df_b.shape[0]
a, b

(0.0038223935353535353, 0.008145527676767678)

In [18]:
z_score = (mean_b - mean_a) / np.sqrt(a+b)
z_score

18.715151117476786

In [19]:
alpha = 0.05

z_critical_value = st.norm.ppf(1 - alpha)
z_critical_value

1.6448536269514722

In [21]:
z_score > z_critical_value

True

In [16]:
p_value = st.norm.cdf(z_score)
p_value

1.8628166660545397e-78

In [17]:
p_value < alpha

True

### Observations and Conclusion

- As you can see above, since the Z score is higher than the critical z value, it falls inside the rejection region. Hence we will reject our null hypothesis and accept the alternate hypothesis that Design B leads to a higher average time spent on the website compared to Design A

## t-test

In [25]:
from scipy.stats import ttest_ind

t_stat, p_value = ttest_ind(df_b["time_spent_minutes"],
                            df_a["time_spent_minutes"],
                            ) 
print(f"T-stat: {t_stat:.3f}, P-value: {p_value:.5f}")
print("Reject H0" if p_value < 0.05 else "Fail to reject H0")


T-stat: 18.715, P-value: 0.00000
Reject H0


Based on the t-test results (T-stat = 18.715, p < 0.00001), we reject the null hypothesis and conclude that users spend significantly more time on the website with Design B compared to Design A. This suggests that the new design successfully increases user engagement.